### PR TITLE
Issue #1629 Documentation of steps for setting up kvm driver on arch linux

### PR DESCRIPTION
**Pull request by anjannath**
addresses issue #1629

### CODE REVIEW

File: docs/source/getting-started/setting-up-driver-plugin.adoc [hunk 1]
```diff
@@ -77,6 +77,33 @@ $ sudo usermod -a -G libvirt <username>
 $ newgrp libvirt
 ----
 
+[[kvm-driver-arch-linux]]
+=== On Arch Linux
```
Comment (hferentschik) [inline docs/source/getting-started/setting-up-driver-plugin.adoc]: What's the specific part? And why do we want to document it explicitly?

Comment (hferentschik) [reply docs/source/getting-started/setting-up-driver-plugin.adoc]: I am using Mint? Can you add instructions for that as well?

Comment (anjannath) [reply docs/source/getting-started/setting-up-driver-plugin.adoc]: we got a lot of questions on the IRC about it also an issue got opened, the specific parts are the two seperate packages that has to be installed and making changes to the `/etc/libvirt/qemu.conf`.

Comment (hferentschik) [reply docs/source/getting-started/setting-up-driver-plugin.adoc]: Sure, if the majority of @minishift/minishift-dev agrees, let's add it. I'm neutral

**Comment by LalatenduMohanty** [comment]
retest this please

File: docs/source/getting-started/setting-up-driver-plugin.adoc (hunk 1 above)
Comment (LalatenduMohanty) [reply docs/source/getting-started/setting-up-driver-plugin.adoc]: I am in favor of adding it as it involves an extra step. I am going to approve this PR. However we should just add the extra step required for Arch Linux and we should not repeat steps. May be we can take another look later to organize this better.

File: docs/source/getting-started/setting-up-driver-plugin.adoc [hunk 2]
```diff
@@ -77,6 +77,33 @@ $ sudo usermod -a -G libvirt <username>
...
++
+----
+$ sudo pacman -S libvirt qemu dnsmasq ebtables
+----
+
+. Add yourself to *kvm* and *libvirt* groups:
++
+----
+$ sudo usemod -a -G kvm,libvirt <username>
+----
```
Comment (LalatenduMohanty) [inline docs/source/getting-started/setting-up-driver-plugin.adoc]: Any specific reason user needs to be added to both `kvm` and `libvirt` groups? For other distribution we only add to libvirt group.

**Comment by LalatenduMohanty** [comment]
Moving to work in progress after talking to @anjannath.

File: docs/source/getting-started/setting-up-driver-plugin.adoc (hunk 1 above)
Comment (gbraad) [reply docs/source/getting-started/setting-up-driver-plugin.adoc]: Linux Mint and Ubuntu are identical in setup, but Arch Linux involves additional steps. This is particularly an important is distro in Asia which is used by our target audience

**Comment by gbraad** [comment]
What is the reason for the WIP?

**Comment by LalatenduMohanty** [comment]
> What is the reason for the WIP?

https://github.com/minishift/minishift/pull/1654#discussion_r148847460

**Comment by gbraad** [comment]
The reason why both `kvm` and `libvirt` group ? ... I can not answer this... I am not an Argh User

**Comment by anjannath** [comment]
I have to find out and check if it is necessary to add to both the groups, right now don't want to break minishift on my system, need to test other stuffs.

File: docs/source/getting-started/setting-up-driver-plugin.adoc [hunk 3]
```diff
@@ -77,6 +77,33 @@ $ sudo usermod -a -G libvirt <username>
...
+. Install *libvirt*, *qemu*, *dnsmasq* and *ebtables* on your system:
++
+----
+$ sudo pacman -S libvirt qemu dnsmasq ebtables
+----
+
+. Add yourself to *kvm* and *libvirt* groups:
++
+----
+$ sudo usemod -a -G kvm,libvirt <username>
```
Comment (praveenkumar) [inline docs/source/getting-started/setting-up-driver-plugin.adoc]: @anjannath for us it something like below, this looks like a long pending PR, can you do a check + rebase so we can merge it?

```
$ sudo groupmems -g kvm -l
qemu 

$ sudo groupmems -g libvirt -l
prkumar 
```

Comment (anjannath) [reply docs/source/getting-started/setting-up-driver-plugin.adoc]: @praveenkumar Sure.. on it now!

Comment (anjannath) [reply docs/source/getting-started/setting-up-driver-plugin.adoc]:
```
➜  ~ sudo groupmems -g kvm -l
anath 
➜  ~ sudo groupmems -g libvirt -l
anath 
```
there is no `qemu` user present in Arch Linux by default.

File: docs/source/getting-started/setting-up-driver-plugin.adoc (hunk 2 above)
Comment (anjannath) [reply docs/source/getting-started/setting-up-driver-plugin.adoc]: checked by reinstalling all the packages and doing the configurations, and it is required to be part of both the groups, but still do not know for sure, why the user needs to be added to both the groups. My guess is that, both `kvm` and `libvirt` groups permission is required for running a kvm virtual machine in Arch Linux.

**Pull request merged**